activerecord/test/cases/relation_test.rb

```diff
@@ -223,6 +224,30 @@ def test_relation_merging_with_merged_symbol_joins_has_correct_size_and_count
       assert_equal manual_comments_on_post_that_have_author.size, merged_authors_with_commented_posts_relation.to_a.size
     end
 
+    def test_relation_merging_with_merged_symbol_joins_is_aliased
+      categorizations_with_authors = Categorization.joins(:author)
+      queries = capture_sql { Post.joins(:author, :categorizations).merge(Author.select(:id)).merge(categorizations_with_authors).to_a }
+
+      nb_inner_join = queries.sum { |sql| sql.scan(/INNER\s+JOIN/i).size }
+      assert_equal 3, nb_inner_join, "Wrong amount of INNER JOIN in query"
+
+      # using `\W` as the column separator
+      assert queries.any? { |sql| %r[INNER\s+JOIN\s+#{Author.quoted_table_name}\s+\Wauthors_categorizations\W]i.match?(sql) }, "Should be aliasing the child INNER JOINs in query"
+    end
+
+    def test_relation_with_merged_joins_aliased_works
```

Comment:
this test status:
- This PR -> pass
- Rails `master` -> ERROR
- Rails `5.2` -> ERROR
- Rails `5.1` -> pass